homeassistant/components/nice_go/coordinator.py

```diff
@@ -178,8 +182,11 @@ async def _update_refresh_token(self) -> None:
 
     async def client_listen(self) -> None:
         """Listen to the websocket for updates."""
-        self.api.event(self.on_connected)
-        self.api.event(self.on_data)
+        self._unsub_connected = self.api.listen("on_connected", self.on_connected)
+        self._unsub_data = self.api.listen("on_data", self.on_data)
+        self._unsub_connection_lost = self.api.listen(
+            "on_connection_lost", self.on_connection_lost
+        )
```

Comment:
When is this called? At line 197 it seems you make a recursive call to `client_listen`, is that correct?

Reply:
Yes but that's only a back up in case something odd happens that causes the library's re-connection logic to fail

Reply:
But it seems not a good idea to do this recursively. Instead it is better to do it in a loop, and after some retries, if it still fail to raise a `ConfigEntryNotReady` instead.

Reply:
I pushed a potential fix, let me know if it looks good